process.py

```diff
@@ -385,12 +386,26 @@ def get_provider_code(directory_name=None):
 
 
 def main():
+    dir_list = []
     parser = argparse.ArgumentParser()
+    parser.add_argument('--specific_folder', dest='specific_folder', action='store_true', default=False)
+    parser.add_argument('--delete_all', dest='delete_all', action='store_true', default=True)
     parser.add_argument('--path', type=str, default="static/data/lms")
     parser.add_argument('--batch_size', type=int, default=100)
-    parser.add_argument('--delete_all', dest='delete_all', action='store_true', default=True)
     parser.add_argument('--provider_code', type=int)
     args = parser.parse_args()
+
+    if args.specific_folder:
+        dirname = tkFileDialog.askdirectory(initialdir=os.path.abspath(args.path),  title='Please select a directory')
+        dir_list.append(dirname)
```

Comment:
You can just do `dir_list = [dirname]` and then you don't need to initialize it to an empty list before